packages/deployments/contracts/src/cli/init/helpers/assets.ts

```diff
@@ -150,5 +153,87 @@ export const setupAsset = async (args: {
         });
       }
     }
+
+    // NOTE: it is best practice to init + add liquidity in a single transaction to start the pool in a balanced state
+    if (apply && +home.chain === 1) {
+      // TODO: add liquidity with balance assertions; proper min to mint calculations; etc.
+      // Fixing this is useful in testnet, but on mainnets youre using safes anyway.
+      console.warn(`Must implement safe pool initialization. Skipping.`);
+      continue;
+    }
+
+    // After registering the asset, check pool status.
+    const [local, adopted] = apply
+      ? await getValue<[string, string]>({
+          deployment: network.deployments.Connext,
+          read: { method: "getLocalAndAdoptedToken(bytes32,uint32)", args: [canonical.id, canonical.domain] },
+        })
+      : [representation.local ?? constants.AddressZero, representation.adopted];
+
+    if (local.toLowerCase() === adopted.toLowerCase()) {
+      // No pools are needed
+      continue;
+    }
+
+    // Get decimals from tokens
+    const tokens = [new Contract(local, ERC20Abi, network.rpc), new Contract(adopted, ERC20Abi, network.rpc)];
+    const decimals = await Promise.all(tokens.map((t) => t.decimals()));
+
+    // Generate inputs for pool
+    const INITIAL_A = 200;
+    const SWAP_FEE = 4e6; // 4bps FEE_DENOMINATOR = 10 ** 10
+    const ADMIN_FEE = 0;
+    const lpTokenName = `${asset.name.toUpperCase()}-${tokenName}LP`;
```

Comment:
@just-a-node is this consistent with what you did with the other LP name / symbol info?

Reply:
`Connext ${asset.symbol} StableSwap LP` but AssetStack type doesn't contain symbol rn

Reply:
we use the symbols as the name in the values provided to the stack